esmf_regrid/schemes.py

```diff
@@ -575,29 +1051,186 @@ def __call__(self, cube):
+            else:
+                dims = cube.coord_dims(src_x)[::-1]
+
+        regrid_info = _RegridInfo(
+            dims=dims,
+            target=self.target,
             regridder=self.regridder,
         )
 
-        return _regrid_rectilinear_to_rectilinear__perform(
-            cube, regrid_info, self.mdtol
+        if cube.mesh is None:
+            if type(self.target[1]) is not str:
+                result = _regrid_rectilinear_to_rectilinear__perform(
+                    cube, regrid_info, self.mdtol
+                )
+            else:
+                result = _regrid_rectilinear_to_unstructured__perform(
+                    cube, regrid_info, self.mdtol
+                )
+        else:
+            if type(self.target[1]) is not str:
+                result = _regrid_unstructured_to_rectilinear__perform(
+                    cube, regrid_info, self.mdtol
+                )
+            else:
+                result = _regrid_unstructured_to_unstructured__perform(
+                    cube, regrid_info, self.mdtol
+                )
+
+        return result
+
+
+class ESMFAreaWeightedRegridder(_ESMFRegridder):
+    r"""Regridder class for unstructured to rectilinear :class:`~iris.cube.Cube`\\ s."""
+
+    def __init__(
+        self,
+        src,
+        tgt,
+        mdtol=0,
+        precomputed_weights=None,
+        resolution=None,
+        srcres=None,
+        tgtres=None,
+        use_src_mask=False,
+        use_tgt_mask=False,
+    ):
+        """
+        Create regridder for conversions between ``src`` and ``tgt``.
+
+        Parameters
+        ----------
+        src : :class:`iris.cube.Cube`
+            The rectilinear :class:`~iris.cube.Cube` providing the source grid.
+        tgt : :class:`iris.cube.Cube`
+            The rectilinear :class:`~iris.cube.Cube` providing the target grid.
+        mdtol : float, default=0
+            Tolerance of missing data. The value returned in each element of
+            the returned array will be masked if the fraction of masked data
+            exceeds ``mdtol``. ``mdtol=0`` means no missing data is tolerated while
+            ``mdtol=1`` will mean the resulting element will be masked if and only
+            if all the contributing elements of data are masked.
+        precomputed_weights : :class:`scipy.sparse.spmatrix`, optional
+            If ``None``, :mod:`esmpy` will be used to
+            calculate regridding weights. Otherwise, :mod:`esmpy` will be bypassed
+            and ``precomputed_weights`` will be used as the regridding weights.
+        resolution : int, optional
+            If present, represents the amount of latitude slices per cell
+            given to ESMF for calculation. If resolution is set, grid_cube
+            must have strictly increasing bounds (bounds may be transposed plus or
+            minus 360 degrees to make the bounds strictly increasing).
+        srcres : int, optional
```

Comment:
Consider [this advice from the Numpy Style guide](https://numpydoc.readthedocs.io/en/stable/format.html#parameters):

> When two or more input parameters have exactly the same type, shape and description, they can be combined:
> 
> ```
> x1, x2 : array_like
>     Input arrays, description of `x1`, `x2`.
> ```

Reply:
see https://github.com/stephenworsley/iris-esmf-regrid/commit/cb27ae6111f23233bad6779279db1205c7f94b46